logprep/framework/pipeline.py

```diff
@@ -344,11 +355,20 @@ def _store_extra_data(self, extra_data: tuple):
 
     def _shut_down(self):
         self._input.shut_down()
+        if hasattr(self._input, "server"):
+            self._used_server_ports.pop(self._input.server.config.port)
+        self._drain_input_queues()
         self._output.shut_down()
-
         while self._pipeline:
             self._pipeline.pop().shut_down()
 
+    def _drain_input_queues(self):
+        if not hasattr(self._input, "_messages"):
+            return
+        if isinstance(self._input._messages, queue.Queue):
+            while self._input._messages.qsize():
```

Comment:
We can use it, because the main cause for this warning is, that it is possible another process puts something in the queue between the call to qsize and the following action. In our case the webserver is down when we use qsize and no one else is able to put something to the queue.